sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/taskdata/TaskLabelWriter.java

```diff
@@ -36,38 +36,25 @@ public TaskLabelWriter(TaskInfo.Builder taskInfoBuilder) {
     }
 
     /**
-     * Ensures that the task is identified as a transient task.
+     * Ensures that the task is identified as a transient task. This is a "task" which is never launched, only sent to
+     * Mesos as a way to capture some resources.
      */
     public TaskLabelWriter setTransient() {
         writer.put(LabelConstants.TRANSIENT_FLAG_LABEL, LabelConstants.BOOLEAN_LABEL_TRUE_VALUE);
         return this;
     }
 
     /**
-     * Ensures that the task is not identified as a transient task.
-     */
-    public TaskLabelWriter clearTransient() {
-        writer.remove(LabelConstants.TRANSIENT_FLAG_LABEL);
-        return this;
-    }
-
-    /**
-     * Ensures that the task is identified as permanently failed.
+     * Ensures that the task is identified as permanently failed. This is intentionally stored in the TaskInfo as it
+     * will be automatically overwritten when the TaskInfo is replaced. As such it will automatically be "cleared" when
+     * the TaskInfo is regenerated in the offer cycle.
      */
     public TaskLabelWriter setPermanentlyFailed() {
         writer.put(LabelConstants.PERMANENTLY_FAILED_LABEL, LabelConstants.BOOLEAN_LABEL_TRUE_VALUE);
         return this;
     }
 
     /**
-     * Ensures that the task is not identified as permanently failed.
-     */
-    public TaskLabelWriter clearPermanentlyFailed() {
```

Comment:
Correct. Rather than explicitly clearing the bit, it's getting overwritten when the new task is launched.